### PR TITLE
Introduce XPRESS compresssion on Windows.

### PR DESCRIPTION
**Pull request by yuslepukhin**
Comparable with Snappy on comp ratio.
  Implemented using Windows API, does not require external package.
  Avaiable since Windows 8 and server 2012.
  Use -DXPRESS=1 with CMake to enable.

### CODE REVIEW

File: tools/sst_dump_tool.cc [hunk 1]
```diff
@@ -193,15 +193,24 @@ int SstFileReader::ShowAllCompressionSizes(size_t block_size) {
...
-  for (CompressionType i = CompressionType::kNoCompression;
-       i <= CompressionType::kZSTDNotFinalCompression;
-       i = (i == kLZ4HCCompression) ? kZSTDNotFinalCompression
-                                    : CompressionType(i + 1)) {
+  CompressionType compressions[] = {
+    CompressionType::kNoCompression,
+    CompressionType::kSnappyCompression,
+    CompressionType::kZlibCompression,
+    CompressionType::kBZip2Compression,
+    CompressionType::kLZ4Compression, // Skip the H type
```
Comment (siying) [inline tools/sst_dump_tool.cc]: I don't think we should skip LZ4HC. It is a good compaction style.

Comment (yuslepukhin) [inline tools/sst_dump_tool.cc]: Ok.

**Comment by yuslepukhin** [comment]
Travis failures are Java related and unrelated to my change

**Comment by yuslepukhin** [comment]
@siying Pls, take a look when you have a chance.

**Pull request merged**